### ISSUE TITLE
Add Shared Projects

### ISSUE THREAD

**Issue by ja2ui0** [pentacent/keila#156]
Unless I'm missing something, projects are specific to the user. Therefore, multiple users managing a project would have to share a login to access the same contacts / segments / campaigns.

It would be useful for multiple users with their own credentials to be able to manage projects collaboratively.

**Comment by wmnnd** [comment]
Keila is already set up to allow this; projects are tied to an entity called "account" and not the user entity. We’re just missing a UI  to manage the shared access to a project.

@ja2ui0: Would you be willing to help with this issue?

**Comment by ja2ui0** [comment]
@wmnnd: I'm not unwilling, it will take me time I don't have at the moment considering other commitments. I'm not a coder or a UI / UX designer, but when things settle down a little bit in a couple of weeks I might be able to have a look.

**Comment by harryfear** [comment]
For now, how can we manually edit the database to enable 2 users to share a project, please?

**Comment by harryfear** [comment]
A row should be added to `user_groups` with the `user_id` (set as per the user to whom to delegate project access to) and the `group_id` (set as per the `group_id` of the Project in question in `projects`). The `id`, `inserted_at` and `updated_at` fields also need setting on insertion.

**Comment by wmnnd** [comment]
@harryfear Do all the other routes work though?

**Comment by harryfear** [comment]
@wmnnd Yes, they do! It's just the on-send route and the stats routes that don't seem to work 😊

**Comment by verymilan** [comment]
Would be amazing if this feature would become an actual thing one day - this actually kinda took me by surprise. 😇 
Would already be helpful if those missing things under the hood would be implemented, as i could get around the missing frontend stuff. :)

**Comment by jlxq0** [comment]
I strongly support this. Would be happy to either have a UI or a supported way on the command line or db.

**Comment by jfayad** [comment]
I also would appreciate see this come to life.

Wondering though, is there a possibility to define permissions when granting access to a project ? (i.e create campaigns but not send them ?)

**Comment by andreialecu** [comment]
Would also like to request being able to create users who cannot see any contact details, but they can just define campaigns.

**Comment by Jimmy-Cl** [comment]
I strongly support this, it would be a really good addition to Keila. 👍

**Comment by Aaron691** [comment]
Any update on if this is coming to light or going to be added to the roadmap?

**Comment by wmnnd** [comment]
Thanks for your suggestion about adding this to the roadmap. I've added it now and this is definitely a feature I want to add to Keila.
